api/server.py

```diff
@@ -115,11 +115,30 @@ async def exception_handler(_: Request, e: Exception):
         st = traceback.format_exc()
         base_params['stacktrace'] = st
 
-    return JSONResponse(
+    response = JSONResponse(
         status_code=code,
         content=base_params,
     )
 
+    # https://github.com/tiangolo/fastapi/issues/457#issuecomment-851547205
+    # FastAPI doesn't run middleware on exception, but if we make a non-GET/INFO
+    # request, then we lose CORS and hence lose the exception in the body of the
+    # response. Grab it manually, and explicitly allow origin if so.
+    middlewares = [m for m in app.user_middleware if isinstance(m, CORSMiddleware)]
```

Comment:
I'm not entirely sure why this is necessary / which case this is addressing. It's only internal infrastructure, so we don't need to be super strict in terms of security, but might maybe still be worth asking someone like Tommy for a review of this part?

Reply:
We could always set it to always? You're sort of prone to scripts on external sites hitting metamist API without the user knowing. It's been working well so far, just unfortunate that the exception handler doesn't run the remaining middleware and hits the CORS middleware (and hence blocks the content body).